applications/ChimeraApplication/custom_processes/apply_chimera_process.cpp

```diff
@@ -0,0 +1,627 @@
+									}],
+									[{
+										"model_part_name":"PLEASE_SPECIFY",
+                                        "search_model_part_name":"PLEASE_SPECIFY",
+                                        "boundary_model_part_name":"PLEASE_SPECIFY",
+										"overlap_distance":0.0
+									}]
+								]
+            })");
+    mNumberOfLevels = mParameters.size();
+    KRATOS_ERROR_IF(mNumberOfLevels < 2)
+        << "Chimera requires atleast one Patch !" << std::endl;
+
+    mEchoLevel = 0;
+    mReformulateEveryStep = false;
+    mIsFormulated = false;
+}
+
+template <int TDim>
+ApplyChimera<TDim>::~ApplyChimera()
+{
+}
+
+template <int TDim>
+void ApplyChimera<TDim>::SetEchoLevel(int EchoLevel)
+{
+    mEchoLevel = EchoLevel;
+}
+
+template <int TDim>
+void ApplyChimera<TDim>::SetReformulateEveryStep(bool Reformulate)
+{
+    mReformulateEveryStep = Reformulate;
+}
+
+template <int TDim>
+void ApplyChimera<TDim>::ExecuteInitializeSolutionStep()
+{
+    KRATOS_TRY;
+    VariableUtils().SetFlag(ACTIVE, true, mrMainModelPart.Elements());
+    // Actual execution of the functionality of this class
+    if (mReformulateEveryStep || !mIsFormulated) {
+        DoChimeraLoop();
+        mIsFormulated = true;
+    }
+    KRATOS_CATCH("");
+}
+
+template <int TDim>
+void ApplyChimera<TDim>::ExecuteFinalizeSolutionStep()
+{
+    VariableUtils().SetFlag(VISITED, false, mrMainModelPart.Nodes());
+    VariableUtils().SetFlag(VISITED, false, mrMainModelPart.Elements());
+    VariableUtils().SetNonHistoricalVariable(SPLIT_ELEMENT, false,
+                                             mrMainModelPart.Elements());
+
+    if (mReformulateEveryStep) {
+        mrMainModelPart.RemoveMasterSlaveConstraintsFromAllLevels(TO_ERASE);
+        mIsFormulated = false;
+    }
+}
+
+template <int TDim>
+std::string ApplyChimera<TDim>::Info() const
+{
+    return "ApplyChimera";
+}
+
+template <int TDim>
+void ApplyChimera<TDim>::PrintInfo(std::ostream& rOStream) const
+{
+    rOStream << "ApplyChimera" << std::endl;
+}
+
+template <int TDim>
+void ApplyChimera<TDim>::PrintData(std::ostream& rOStream) const
+{
+    rOStream << "ApplyChimera" << std::endl;
+}
+
+template <int TDim>
+void ApplyChimera<TDim>::DoChimeraLoop()
+{
+    const int num_elements = static_cast<int>(mrMainModelPart.NumberOfElements());
+    const auto elem_begin = mrMainModelPart.ElementsBegin();
+
+    Parameters parameters_for_validation(R"(
+                                {
+                                    "model_part_name":"PLEASE_SPECIFY",
+                                    "search_model_part_name":"PLEASE_SPECIFY",
+                                    "boundary_model_part_name":"PLEASE_SPECIFY",
+                                    "model_import_settings":{},
+                                    "overlap_distance":0.0,
+                                    "internal_parts_for_chimera":[]
+                                }
+        )");
+
+#pragma omp parallel for
```

Comment:
```suggestion
#pragma omp parallel for firstprivate(elems_begin)
```
I think however that this is the default behaviour in any case. @RiccardoRossi @philbucher @roigcarlo  can you confirm?

Reply:
`elems_begin` is a `const`  so i think it does not matter.

Reply:
Agree with @adityaghantasala. In general if the variable is not written no need to specify if its is shard or private. (but if you want to make it private for some reason it has to be firstprivate)